### PR TITLE
Updated client library to use ingest v2/event end point for all events

### PR DESCRIPTION
**Pull request by sheremetat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,7 +110,11 @@ client.send({
 
 Events can be sent to SignalFx via the `sendEvent` function. The
 event type must be specified, and dimensions and extra event properties
-can be supplied as well.
+can be supplied as well. Also please specify event category: for that get
+option from dictionary `client.EVENT_CATEGORIES`. Different categories of events are supported.
+Available categories of events are `USER_DEFINED`, `ALERT`, `AUDIT`, `JOB`,
```
Comment (unsetbit) [inline README.md]: I think the only one we want people to use here is USER_DEFINED. Could we make this an implied default with a way to override?

File: lib/client/json_signal_fx_client.js [hunk 2]
```diff
@@ -44,4 +44,23 @@ JsonSignalFx.prototype._batchData = function (datapointsList) {
...
+JsonSignalFx.prototype._buildEvent = function (event) {
+  var trimedEvent = {};
+  for (var key in event) {
+    if (event.hasOwnProperty(key)) {
+      if (event[key]) {
+        trimedEvent[key] = event[key];
+      }
+    }
+  }
+  var eventsListToSend = [];
```
Comment (unsetbit) [inline lib/client/json_signal_fx_client.js]: eventsListToSend = [trimmedEvent]

File: lib/client/json_signal_fx_client.js [hunk 3]
```diff
@@ -44,4 +44,23 @@ JsonSignalFx.prototype._batchData = function (datapointsList) {
   return JSON.stringify(data);
 };
 
+JsonSignalFx.prototype._buildEvent = function (event) {
+  var trimedEvent = {};
```
Comment (unsetbit) [inline lib/client/json_signal_fx_client.js]: spelling should be trimmedEvent

File: lib/client/signal_fx_client.js [hunk 4]
```diff
@@ -146,16 +155,35 @@ SignalFxClient.prototype._batchData = function (datapointsList) {
...
  */
-SignalFxClient.prototype.sendEvent = function (eventType, dimensions, properties) {
+SignalFxClient.prototype.sendEvent = function (eventCategory, eventType, dimensions, properties, timestamp) {
   var _this = this;
+  if (!eventType) {
+    throw new Error('Type of event should not be empty!');
+  }
+  // Check than passed event category is supported
+  var supportedEventCategoriesValues = [];
+  for (var key in this.EVENT_CATEGORIES) {
```
Comment (unsetbit) [inline lib/client/signal_fx_client.js]: This is a static list, shouldn't be reconstructing it with every sendEvent call

**Comment by sheremetat** [comment]
Done. Please review.

File: example/general_usage.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ var signalFx = require('../lib/signalfx');
 var token = 'YOUR SIGNALFX TOKEN'; // Replace with you token
 
 var client = new signalFx.SignalFx(token, {
-  enableAmazonUniqueId: true, // Retrieve and add Amazon unique identifier as dimension
+  enableAmazonUniqueId: false, // Retrieve and add Amazon unique identifier as dimension
```
Comment (unsetbit) [inline example/general_usage.js]: since you changed this to false, the comment no longer applied

**Comment by unsetbit** [comment]
lgtm after the comment fix

**Comment by sheremetat** [comment]
Thanks for review! Comment fixed, commits squashed. Ready for merge.

**Pull request merged**